common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/requests/NativeAuthRequest.kt

```diff
@@ -0,0 +1,43 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+
+package com.microsoft.identity.common.java.providers.nativeauth.requests
+
+import org.json.JSONObject
+import java.net.URL
+
+abstract class NativeAuthRequest {
+    abstract var requestUrl: URL
+    abstract var headers: Map<String, String?>
+    abstract val parameters: NativeAuthRequestParameters
+
+    abstract class NativeAuthRequestParameters {
```

Comment:
Javadoc

Reply:
You may want to move this to its own file

Reply:
I will add the javadoc. I cannot move the file to its own file as the NativeAuth class encloses NativeAuthRequestParameters.